### PR TITLE
Spritemap Improvements

### PR DESCRIPTION
**Pull request by MattTuttle**
Various improvements to the Spritemap class.

* Moved `Animation` into Spritemap file to "hide" it from external use
* Removed `setAnimFrame`. This indirectly updated the frame, where the animation frame chooses from the current animations frames. I'm not sure anyone uses this.
* Changed `currentAnimation`. Instead of a String it holds the current `Animation` object.
  * This variable also uses the null safe class `Maybe<T>`. Thanks nadako!
* Changed the callback function for animation end to a singal named `endAnimation`.
* Added the ability to load a TileAtlas from a packed texture. It requires and ordered list of names.
* Width/Height reflect the current frame (packed texture can load different size regions).

### CODE REVIEW

File: haxepunk/graphics/atlas/TileAtlas.hx [hunk 1]
```diff
@@ -28,10 +33,29 @@ class TileAtlas extends Atlas
...
-	
+
+	/**
+	 * Loads a TexturePacker xml file and generates all tile regions.
+	 * Uses the Generic XML exporter format from Texture Packer.
+	 * @param	file	The TexturePacker file to load
+	 * @param	sprites	A list of the sprite names in the packed file
+	 * @return	A TileAtlas with all packed images defined as regions ordered by sprite names supplied
+	 */
+	public static function loadTexturePacker(file:String, sprites:Array<String>):TileAtlas
```
Comment (bendmorris) [inline haxepunk/graphics/atlas/TileAtlas.hx]: This only supports one specific TextureAtlas format. Would it makes sense instead to just pass in any existing TextureAtlas?

**Comment by bendmorris** [comment]
Nice, `Maybe<T>` seems potentially useful all over the place ("maybe" everywhere we're currently using `Null<T>`.) Seems like something that would make sense in the official `haxe.ds`.

**Comment by bendmorris** [comment]
Strangely Haxe provides `Option` (https://haxe.org/manual/std-Option.html) which is conceptually the same but not as useful as Maybe and imposes a runtime cost. Maybe seems strictly better.

File: haxepunk/graphics/atlas/TileAtlas.hx (hunk 1 above)
Comment (MattTuttle) [inline haxepunk/graphics/atlas/TileAtlas.hx]: That's a good point. I'm not sure what the appropriate name would be for this function.

**Comment by MattTuttle** [comment]
Option doesn't force you to unwrap an object though. I really like Swift's null safety and this gets a bit closer with very little runtime overhead.

**Comment by MattTuttle** [comment]
The codeclimate issues are all related to inline functions. These look better with Egyptian style braces but I can change them if you think the braces should be on individual lines.

**Pull request merged**